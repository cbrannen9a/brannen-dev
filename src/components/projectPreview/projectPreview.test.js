import React from 'react';
import renderer from 'react-test-renderer';

import ProjectPreview from './projectPreview';

const mock = {
    title: 'Test Project',
    slug: 'testslug',
    imageData: {
        aspectRatio: 1.5,
        src: 'test_image.jpg',
        srcSet: 'some srcSet',
        srcSetWebp: 'some srcSetWebp',
        sizes: '(max-width: 600px) 100vw, 600px',
        base64: 'string_of_base64'
    },
    description: 'test description',
    skills: [
        {
            id: 'skill1',
            title: 'isSkill1',
            _rawDescription: [
                {
                    style: 'normal',
                    _key: 'block1',
                    _type: 'block',
                    markDefs: [],
                    children: [
                        {
                            marks: [],
                            text: 'test text',
                            _key: 'mock1',
                            _type: 'span'
                        },
                        {
                            marks: ['strong'],
                            text: 'test bold',
                            _key: 'mock2',
                            _type: 'span'
                        },
                        {
                            marks: ['em'],
                            text: 'test italic',
                            _key: 'mock3',
                            _type: 'span'
                        }
                    ]
                }
            ]
        },
        {
            id: 'skill2',
            title: 'isSkill2',
            _rawDescription: [
                {
                    style: 'normal',
                    _key: 'block1',
                    _type: 'block',
                    markDefs: [],
                    children: [
                        {
                            marks: [],
                            text: 'test text',
                            _key: 'mock1',
                            _type: 'span'
                        },
                        {
                            marks: ['strong'],
                            text: 'test bold',
                            _key: 'mock2',
                            _type: 'span'
                        },
                        {
                            marks: ['em'],
                            text: 'test italic',
                            _key: 'mock3',
                            _type: 'span'
                        }
                    ]
                }
            ]
        }
    ]
};

describe('ProjectPreview', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<ProjectPreview {...mock} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
