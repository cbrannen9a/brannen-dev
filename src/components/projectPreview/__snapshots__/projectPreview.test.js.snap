// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectPreview renders correctly 1`] = `
<div
  className="project-preview"
>
  <a
    href="/testslug/"
  >
    <div
      className=" gatsby-image-wrapper"
      style={
        Object {
          "overflow": "hidden",
          "position": "relative",
        }
      }
    >
      <div
        style={
          Object {
            "paddingBottom": "66.66666666666667%",
            "width": "100%",
          }
        }
      />
      <img
        alt=""
        src="string_of_base64"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "objectFit": "cover",
            "objectPosition": "center",
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transitionDelay": "500ms",
            "width": "100%",
          }
        }
      />
      <picture>
        <source
          sizes="(max-width: 600px) 100vw, 600px"
          srcSet="some srcSetWebp"
          type="image/webp"
        />
        <source
          sizes="(max-width: 600px) 100vw, 600px"
          srcSet="some srcSet"
        />
        <img
          alt="Test Project"
          loading="lazy"
          onLoad={[Function]}
          sizes="(max-width: 600px) 100vw, 600px"
          src="test_image.jpg"
          srcSet="some srcSet"
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 0,
              "position": "absolute",
              "top": 0,
              "transition": "opacity 500ms",
              "width": "100%",
            }
          }
        />
      </picture>
      <noscript
        dangerouslySetInnerHTML={
          Object {
            "__html": "<picture><source type='image/webp' srcset=\\"some srcSetWebp\\" sizes=\\"(max-width: 600px) 100vw, 600px\\" /><source srcset=\\"some srcSet\\" sizes=\\"(max-width: 600px) 100vw, 600px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 600px) 100vw, 600px\\" srcset=\\"some srcSet\\" src=\\"test_image.jpg\\" alt=\\"Test Project\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
          }
        }
      />
    </div>
  </a>
  <h2>
    <a
      href="/testslug/"
    >
      Test Project
    </a>
  </h2>
  <a
    className="button is-link"
    href="/testslug/"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-eye fa-w-18 "
      data-icon="eye"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={
        Object {
          "marginRight": "3px",
        }
      }
      viewBox="0 0 576 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    View
  </a>
  <p>
    test description
  </p>
  <span
    className="tags"
    style={
      Object {
        "marginTop": "5px",
      }
    }
  >
    <div
      style={
        Object {
          "marginRight": "4px",
        }
      }
    >
      <span
        className="tooltip"
        onMouseLeave={[Function]}
      >
        <span
          className="tooltip-trigger"
          onMouseOver={[Function]}
        >
          <span
            className="tag"
          >
            isSkill1
          </span>
        </span>
      </span>
    </div>
    <div
      style={
        Object {
          "marginRight": "4px",
        }
      }
    >
      <span
        className="tooltip"
        onMouseLeave={[Function]}
      >
        <span
          className="tooltip-trigger"
          onMouseOver={[Function]}
        >
          <span
            className="tag"
          >
            isSkill2
          </span>
        </span>
      </span>
    </div>
  </span>
</div>
`;
